pkg/apiserver/rest/dualwriter_mode2_test.go

```diff
@@ -607,3 +625,57 @@ func TestEnrichReturnedObject(t *testing.T) {
 		})
 	}
 }
+
+func TestMode2_Sync(t *testing.T) {
```

Comment:
Nice. We might want (not necessarily in this PR) to expand integration testing to cover syncing as well: https://github.com/grafana/grafana/blob/main/pkg/tests/apis/playlist/playlist_test.go